include/ap/argument_parser.hpp

```diff
@@ -268,121 +487,186 @@ class optional_argument : public argument_interface {
+     * @tparam A Valid argument specifier type (positional or optional).
+     * @param name The primary name of the argument.
+     * @param short_name The short name of the argument (optional).
+     * @return Reference to the added argument.
+     */
     template <detail::valid_argument_specifier A>
     detail::argument_type<A>&
         add_argument(const std::string_view name, const std::string_view short_name);
 
-    friend std::ostream& operator<< (std::ostream& os, const argument_parser& parser) {
-        if (parser._program_name)
-            os << parser._program_name.value() << std::endl;
-
-        if (parser._program_description)
-            os << parser._program_description.value() << std::endl;
-
-        return os;
-    }
+    /**
+     * @brief Output stream operator to print the program name and description.
+     * @param os Output stream.
+     * @param parser Argument parser to be printed.
+     * @return Reference to the output stream.
+     */
+    friend std::ostream& operator<< (std::ostream& os, const argument_parser& parser);
 
 private:
-    void _check_arg_name_present(const std::string_view name) const {
-        const auto name_eq_predicate = [name](const detail::argument_interface& arg) {
-            return name == arg.name();
-        };
-
-        if (std::find_if(
-                this->_positional_args.begin(), this->_positional_args.end(), name_eq_predicate
-            ) != this->_positional_args.end()) {
-            throw std::invalid_argument(
-                "Argument name [" + std::string(name) + "] invalid! Name already used in another argument"
-            );
-        }
-
-        if (std::find_if(
-                this->_optional_args.begin(), this->_optional_args.end(), name_eq_predicate
-            ) != this->_optional_args.end()) {
-            throw std::invalid_argument(
-                "Argument name [" + std::string(name) + "] invalid! Name already used in another argument"
-            );
-        }
-    }
-
-    std::optional<std::string_view> _program_name;
-    std::optional<std::string_view> _program_description;
-
-    std::vector<detail::positional_argument> _positional_args;
-    std::vector<detail::optional_argument> _optional_args;
+    /**
+     * @brief Checks if the argument name is already present in the parser.
+     * @param name The argument name to be checked.
+     * @throws std::invalid_argument if the name is already used in another argument.
+     */
+    void _check_arg_name_present(const std::string_view name) const;
+
+    std::optional<std::string_view> _program_name;           ///< Optional program name.
+    std::optional<std::string_view> _program_description;    ///< Optional program description.
+
+    std::vector<detail::positional_argument> _positional_args;    ///< Vector of positional arguments.
+    std::vector<detail::optional_argument> _optional_args;        ///< Vector of optional arguments.
 };
 
+/**
```

Comment:
No need for doc comments here - these are template specializations of the `argument_parser::add_argument` functions